internal/common/retry/interceptor.go

```diff
@@ -0,0 +1,156 @@
+	// Should not be used together with unlimited attempts as resulting retry interval can grow to unreasonable values.
+	UnlimitedInterval = 0
+	// DefaultBackoffCoefficient is default backOffCoefficient for retryPolicy
+	DefaultBackoffCoefficient = 2.0
+	// DefaultMaximumInterval is default maximum amount of time for an individual retry.
+	DefaultMaximumInterval = 10 * time.Second
+	// DefaultExpirationInterval is default expiration time for all retry attempts.
+	DefaultExpirationInterval = time.Minute
+	// DefaultMaximumAttempts is default maximum number of attempts.
+	DefaultMaximumAttempts = UnlimitedMaximumAttempts
+	// DefaultJitter is a default jitter applied on the backoff interval for delay randomization.
+	DefaultJitter = 0.2
+)
+
+type (
+	// GrpcRetryConfig defines required configuration for exponential backoff function that is supplied to gRPC retrier.
+	GrpcRetryConfig struct {
+		initialInterval    time.Duration
+		backoffCoefficient float64
+		maximumInterval    time.Duration
+		expirationInterval time.Duration
+		jitter             float64
+		maximumAttempts    int
+	}
+
+	contextKey string
+)
+
+func (ck contextKey) String() string {
+	return string(ck)
+}
+
+// SetBackoffCoefficient sets rate at which backoff coefficient will change.
+func (g *GrpcRetryConfig) SetBackoffCoefficient(backoffCoefficient float64) {
+	g.backoffCoefficient = backoffCoefficient
+}
+
+// SetMaximumInterval defines maximum amount of time between attempts.
+func (g *GrpcRetryConfig) SetMaximumInterval(maximumInterval time.Duration) {
+	g.maximumInterval = maximumInterval
+}
+
+// SetExpirationInterval defines total amount of time that can be used for all retry attempts.
+// Note that this value is ignored if deadline is set on the context.
+func (g *GrpcRetryConfig) SetExpirationInterval(expirationInterval time.Duration) {
+	g.expirationInterval = expirationInterval
+}
+
+// SetJitter defines level of randomization for each delay interval. For example 0.2 would mex target +- 20%
+func (g *GrpcRetryConfig) SetJitter(jitter float64) {
+	g.jitter = jitter
+}
+
+// SetMaximumAttempts defines maximum total number of retry attempts.
+func (g *GrpcRetryConfig) SetMaximumAttempts(maximumAttempts int) {
+	g.maximumAttempts = maximumAttempts
+}
+
+// NewGrpcRetryConfig creates new retry config with specified initial interval and defaults for other parameters.
+// Use SetXXX functions on this config in order to customize values.
+func NewGrpcRetryConfig(initialInterval time.Duration) *GrpcRetryConfig {
+	return &GrpcRetryConfig{
+		initialInterval:    initialInterval,
+		backoffCoefficient: DefaultBackoffCoefficient,
+		maximumInterval:    DefaultMaximumInterval,
+		expirationInterval: DefaultExpirationInterval,
+		jitter:             DefaultJitter,
+		maximumAttempts:    DefaultMaximumAttempts,
+	}
+}
+
+const (
+	// ConfigKey context key for GrpcRetryConfig
+	ConfigKey = contextKey("RetryConfig")
+)
```

Comment:
```suggestion
varr (
	retryConfigContextKey contextKey
)
```

Reply:
This is encapsulation mehanizm. No one outside of this package won't be able to get value from context because key is unexported value of unexported type.